tests/integration/test_calico.py

```diff
@@ -0,0 +1,191 @@
+
+APISERVER_SPEC = r"""
+# This section configures the Calico API server.
+# For more information, see:
+# https://docs.tigera.io/calico/latest/reference/installation/api
+apiVersion: operator.tigera.io/v1
+kind: APIServer
+metadata:
+  name: default
+spec: {}
+"""
+
+
+def get_installation_spec(registry, repo):
+    return f"""
+# This section includes base Calico installation configuration.
+# For more information, see: https://docs.tigera.io/calico/latest/reference/installation/api#operator.tigera.io/v1.Installation
+apiVersion: operator.tigera.io/v1
+kind: Installation
+metadata:
+  name: default
+spec:
+  # Image format:<registry>/<imagePath>/<imagePrefix><imageName>:<image-tag>
+  registry: {registry}
+  imagePath: "{repo}"
+  imagePrefix: calico-
+
+  # Configures Calico networking.
+  calicoNetwork:
+    ipPools:
+    - name: default-ipv4-ippool
+      blockSize: 26
+      cidr: 192.168.0.0/16
+      encapsulation: VXLANCrossSubnet
+      natOutgoing: Disabled
+      nodeSelector: !all()
+"""
+
+
+def get_image_sha256_digest(image):
+    cmd = ["docker", "manifest", "inspect", image, "-v"]
+    proc = subprocess.run(cmd, check=True, capture_output=True)
+    out_json = json.loads(proc.stdout.decode("utf8"))
+    return out_json["Descriptor"]["digest"]
+
+
+def get_imageset_spec(operator_version, calico_version):
+    spec = {
+        "apiVersion": "operator.tigera.io/v1",
+        "kind": "ImageSet",
+        "metadata": {"name": f"calico-{calico_version}"},
+        "spec": {"images": []},
+    }
+
+    # The Calico operator will error out if we pass other images than the ones
+    # that it expects.
+    calico_images = [
+        "calico-node",
+        "calico-cni",
+        "calico-typha",
+        "calico-kube-controllers",
+        "calico-csi",
+        "calico-apiserver",
+        "calico-ctl",
+        "calico-pod2daemon-flexvol",
+        "calico-key-cert-provisioner",
+        "calico-node-driver-registrar",
+    ]
+    images = list(zip(calico_images, [calico_version] * len(calico_images)))
+    images.append(("calico-tigera-operator", operator_version))
+
+    for image, version in images:
+        rock = env_util.get_build_meta_info_for_rock_version(
+            image, version, IMG_PLATFORM
+        )
+        sha256_digest = get_image_sha256_digest(rock.image)
+        prefix = rock.image.split("/")[1]
+
+        spec["spec"]["images"].append(
+            {
+                "image": f"{prefix}/{image}",
+                "digest": sha256_digest,
+            }
+        )
+
+    return json.dumps(spec)
+
+
+def parse_image(image):
+    # We expect the image to look like this:
+    # ghcr.io/$repo/$name:$tag
+    parts = image.split("/")
+    return {
+        "registry": parts[0],
+        "repo": parts[1],
+        "name": parts[2].split(":")[0],
+        "tag": parts[2].split(":")[1],
+    }
+
+
+@pytest.mark.parametrize("operator_version,calico_version", [("v1.34.0", "v3.28.0")])
+def test_calico(
```

Comment:
I believe we are not disabling the network feature of the k8s-snap which is cilium by default, I feel running 2 CNIs could lead to false positives. Should we disable the feature with a bootstrap config?

We will be testing these images more extensively when moonray branch switches the images.

Reply:
Right, I didn't want to mingle with the k8s snap bootstrap process, so I set Calico to be a no-op:

```
apiVersion: operator.tigera.io/v1
 kind: Installation
   ...
   calicoNetwork:
     ipPools:
     - name: default-ipv4-ippool
       blockSize: 26
       cidr: 192.168.0.0/16
       encapsulation: VXLANCrossSubnet
       natOutgoing: Disabled
       nodeSelector: !all()
```